app/src/main/java/com/sanjeevyadavit/magecart/viewmodel/LoginViewModel.kt

```diff
@@ -0,0 +1,69 @@
+package com.sanjeevyadavit.magecart.viewmodel
+
+import android.app.Application
+import android.util.Log
+import android.widget.Toast
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.sanjeevyadavit.magecart.model.LoginBodyRequest
+import com.sanjeevyadavit.magecart.service.ApiInterface
+import com.sanjeevyadavit.magecart.service.RetrofitClient
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import retrofit2.Call
+import retrofit2.Response
+
+class LoginViewModel(app: Application): AndroidViewModel(app) {
+    val email: MutableLiveData<String> = MutableLiveData("")
+    val password: MutableLiveData<String> = MutableLiveData("")
+
+    private val _customerToken = MutableLiveData<String?>(null)
+    val customerToken: MutableLiveData<String?>
+        get() = _customerToken
+
+    fun login(){
+        if(handleEmptyValues()) return
+        makeLoginApiCall()
+    }
+
+    private fun handleEmptyValues(): Boolean{
+        if(email.value!!.isEmpty()  || password.value!!.isEmpty()) {
+           showToast("Fields cannot be empty")
+            return true;
+        }
+        return false
+    }
+
+    private fun makeLoginApiCall(){
+        viewModelScope.launch(Dispatchers.IO) {
+            val facade: ApiInterface = RetrofitClient.getInstance().create(ApiInterface::class.java)
```

Comment:
You should create a class like the way you did for `Retrofit` instance to get `ApiInterface` as well.

Reply:
Didn't understand